1-src/1-api/2-auth/auth-utilities.mts

```diff
@@ -1,34 +1,52 @@
 import dotenv from 'dotenv';
 import { NextFunction, Request, Response } from 'express';
-import JWT_PKG, { JwtPayload } from 'jsonwebtoken';
+import jwtPackage, { JwtPayload } from 'jsonwebtoken';
 import { createHash } from 'node:crypto';
 import { RoleEnum } from '../../0-assets/field-sync/input-config-sync/profile-field-config.mjs';
 import USER from '../../2-services/1-models/userModel.mjs';
 import { DB_SELECT_USER, DB_SELECT_USER_PROFILE } from '../../2-services/2-database/queries/user-queries.mjs';
 import * as log from '../../2-services/log.mjs';
 import { JwtData } from './auth-types.mjs';
 import { LoginResponseBody } from '../../0-assets/field-sync/api-type-sync/auth-types.mjs';
+import { GetSecretValueCommand, GetSecretValueResponse, SecretsManagerClient } from '@aws-sdk/client-secrets-manager';
 dotenv.config(); 
-const {sign, verify, decode} = JWT_PKG;
 
 /********************
    Create secret key
 ********************/
-var APP_SECRET_KEY;
+let APP_SECRET_KEY:string | undefined;
 
-export const generateSecretKey = () => {
-   const time = new Date().getTime();
-   APP_SECRET_KEY = createHash('sha256').update(time + process.env.SECRET_KEY).digest('base64');
+const getJWTSecretValue = async ():Promise<string> => {
+    try {
+        const client = new SecretsManagerClient({ region: process.env.JWT_SECRET_REGION });
+        const response:GetSecretValueResponse = await client.send(new GetSecretValueCommand({
+            SecretId: process.env.JWT_SECRET_NAME
+        }));
+        return JSON.parse(response.SecretString);
+    } catch (error) {
+        await log.alert(`JWT | AWS Secret Manager failed to connect to JWT Secret: ${process.env.JWT_SECRET_NAME} in Region: ${process.env.JWT_SECRET_REGION}.`, error);
+        throw error;
+    }
+}
+
+export const InitializeJWTSecretKey = async ():Promise<string> => {
+    if(process.env.ENVIRONMENT === 'PRODUCTION') {
+        APP_SECRET_KEY = await getJWTSecretValue();
+    }
+    else APP_SECRET_KEY = process.env.SECRET_KEY;
+    return APP_SECRET_KEY;
 }
-generateSecretKey();
+
+await InitializeJWTSecretKey();
 
 /* *******************
  JWT Token Management
 ******************* */
 export const generateJWT = (userID:number, userRole:RoleEnum):string => {
     //generate JWT as type JWTData
     if(userID > 0 && userRole as RoleEnum !== undefined)
-        return sign({jwtUserID: userID, jwtUserRole: userRole}, APP_SECRET_KEY, {expiresIn: '2 days'});
+        // default config generates signature with HS256 - https://www.npmjs.com/package/jsonwebtoken
+        return jwtPackage.sign({jwtUserID: userID, jwtUserRole: userRole}, APP_SECRET_KEY, {expiresIn: '2 days'});
```

Comment:
While we're doing this, let's set `expiresIn` to an env variable; maybe if undefined fallback to little less like `2 hours`.

Reply:
Just saw your planning to use `KMS's GenerateDataKe`

Reply:
I added a commit to this branch